### ISSUE TITLE
Encoding overloaded after AMD driver 17.10.2

### ISSUE THREAD

**Issue by meredith2u** [Xaymar/OBS-AMD-Advanced-Media-Framework#288]
Recently I upgraded my operating system to Windows 10 1709 version and encoding is overloaded with plugins 2.2.4, H264 and HEVC both happen, but when I using Relive to recording is no issues.

Here is log:
https://gist.github.com/anonymous/84e9219db5c0da66986ed6ccc5a6c7e7

**Comment by jonipokka** [comment]
I had the exact same problem. I have an RX580 and Ryzen 5 1600 cpu with 16 gigs of 3200MHz ram.

Could not record any game without dropping large amount of frames. Happened more on hardware demanding games like Destiny 2. Framerate was capped to 75 since I'm using freesync anyway and the games always ran flawless.

Ultimately gave up trying to fix the issue by changing every possible setting and installed older version of the AMD drivers (17.7.2) and OBS which fixed the issue.

The issues started when the plugin and obs got updated after the AMD driver update which introduced the Failed to start recording issue. Prior to that everything ran flawlessly.

**Comment by meredith2u** [comment]
My previous operating system is Windows 10 1607, plugins 2.2.4 no problem, maybe because Windows 10 1709 changed some of the behavior?
Also, AMD driver versions are using 17.11.2, OBS version is 20.1.1

**Comment by CKannas** [comment]
Hi all,

I think you have to disable Game Mode...

Best,

Christos Kannas

**Comment by meredith2u** [comment]
I'm always disabled Game Mode.

**Comment by Benman2785** [comment]
did it happens when you disable prepass?

**Comment by meredith2u** [comment]
No, It's always happen, no matter prepass turn on/off. I use The Witcher 3 for the test, It seems to only happen in games that require high performance, but on Windows 10 1607 was no issues.

**Comment by meredith2u** [comment]
Today, I just change my  operating system to Windows 10 1607 version and play The Witcher 3, H264 and HEVC encoding all fine.

**Comment by Xaymar** [comment]
I'm on Windows 10 Version 1709 Fall Creators Update and can't confirm any of this. Please actually verify that you disabled Game Bar & Game DVR completely:

- https://www.addictivetips.com/windows-tips/disable-game-mode-in-fall-creators-update-windows-10/
- http://www.tomshardware.co.uk/faq/id-3091550/turn-game-bar-windows.html

**Comment by Xaymar** [comment]
> @jonipokka The issues started when the plugin and obs got updated after the AMD driver update which introduced the Failed to start recording issue. Prior to that everything ran flawlessly.

Could you explain what you mean by this?

**Comment by meredith2u** [comment]
Yes, I did disabled Game mode & Game DVR for the test.

**Comment by meredith2u** [comment]
I did double check, make sure Game mode & Game DVR all disable and upgrade my operating system to Windows 10 Version 1709 then test again, It's same issues.

**Comment by jonipokka** [comment]
@Xaymar 

_20.1.1 Hotfix Changes (Windows only)_

_Made a workaround for an issue with new AMD GPU drivers (ReLive 17.10+) that would prevent the AMD encoder from starting up_

This was the update that broke the encoding for me.

Also just a few days ago I installed a fresh Windows 10 pro on my desktop which was running the Windows version 1706 before the re-install which is now updated to the fall creators update 1709.

Running AMD driver version 17.11.2 and finally had the time to install and test OBS today and it seemed to run fine and not drop any frames.

Oh and forgot to mention that Game mode and Game DVR were both disabled by editing the registry.

**Comment by meredith2u** [comment]
When I playing This war of mine or not play game, H264/HEVC encoding just fine, but when I playing require high performance game, such as The Witcher 3, Encoding overloaded just happen.

**Comment by Xaymar** [comment]
> @jonipokka: This was the update that broke the encoding for me.

Well that's simply impossible, because all that update did is remove the querying of completely unused and removed properties from the new driver: https://github.com/Xaymar/obs-studio_amf-encoder-plugin/commit/d09521eaf4d771bf0d89622f4982932168b070e5

> @meredith2u When I playing This war of mine or not play game, H264/HEVC encoding just fine, but when I playing require high performance game, such as The Witcher 3, Encoding overloaded just happen.

Please make sure that you leave enough GPU time for OBS to compose the frame, download it to RAM, convert it (if necessary) and then upload it. This is the overhead you have to take into account when using OBS instead of ShadowPlay or ReLive. Generally you should aim to get 1.5 times your monitors refresh rate in FPS in the most GPU taxing area of the game, and then play with VSync enabled.

**Comment by jonipokka** [comment]
@Xaymar 
Impossible or not this is when the issues started happening.

I'm not pointing finger at the encoder itself you see, I'm simply stating that one or a combination of the following caused the issues: Updating the drivers to 17.10.X+, updating OBS to the latest version at the time and updating the encoder.

And regarding the headroom thing. I have played games that taxed my GPU and CPU to the limit and still recorded with OBS with zero issues. If the game is capped to 75 and at times it drops to 50 because hardware can't keep up in no case should the recording become unusable due to encoder overload causing **several seconds to be missing from the video**. The video should drop to 50 frames at those times just like the game.

Even with the best hardware it is completely utopian to expect the game to never have minor dips due to loading assets, excess particle effects, problems with the game itself etc and in these cases the hardware is taxed to the limit and the headroom is gone so it is ok and normal in these cases for frames to be completely gone for several seconds in these situations?

Anyway the problem seems to be gone for me now and I just wanted to comment on this thread to let you know that op was not the only one with this issue and **this has never been a problem before.**

**Comment by meredith2u** [comment]
@Xaymar 
On The Witcher 3, I have tried limit fps to 60 and turn vsync on, It still happen.
I'm not sure about your suggest, because I'm using same AMD Driver version and game setting, It only happen on Windows 10 version 1709, version 1607 no issues.

**Comment by Xaymar** [comment]
> @jonipokka: And regarding the headroom thing. I have played games that taxed my GPU and CPU to the limit and still recorded with OBS with zero issues. If the game is capped to 75 and at times it drops to 50 because hardware can't keep up in no case should the recording become unusable due to stuttering and several seconds missing from the video. The video should drop to 50 frames at those times just like the game.
> 
> Even with the best hardware it is completely utopian to expect the game to never have minor dips due to loading assets, excess particle effects, problems with the game itself etc and in these cases the hardware is taxed to the limit and the headroom is gone so it is ok and normal in these cases for frames to be completely gone for several seconds in these situations?

If this dip is caused by the GPU being used completely by the game, how is OBS or any other GPU relying application supposed to work? The focused application always has GPU priority, there is nothing that I can do about it. This is not a case of "I won't do it", it's straight up a case of **"I can't do it because it's not possible!"**. There is simply not enough GPU resources left for:

1. Composing the frame on the GPU, which is done by OBS Studio.
2. Downloading the frame from the GPU, which is done by OBS Studio.
3. Converting the frame, which is done by OBS Studio.
4. Attaching all necessary data for encoding, which is done by the Plugin.
5. Uploading the frame to the GPU, which is done by the Plugin and the AMD Driver AMF Runtime.
6. Signalling the Encoding chip on the GPU, which is done by the AMD Driver AMF Runtime.
7. Downloading the encoded packet back from the GPU, which is done by the Plugin and the AMD Driver AMF Runtime.

Do you see the problem yet? If not, here's a simpler example:

> Imagine you are working, and your boss drops a stack of papers that he wants signed in an hour. Easily enough, you go through all the papers, verify them and sign them in just under an hour and still have some time left for other things.  
> 
> The next day your boss comes in and drops a stack of papers three times that size on your desk, but you still have the same time restriction. You try to work faster (Boost clock), but you can't keep up, so after this hour is over, you still have about half the stack left.
> 
> Your boss is seriously pissed at you and fires you for underperforming.

And this is exactly what you're doing, in this case you're the boss and you're asking the impossible of the worker (gpu). I've done everything I can on the plugin end, now it's time for you to also do your part and balance the load properly. The plugin already employs a frame queue, 

> Anyway the problem seems to be gone for me now and I just wanted to comment on this thread to let you know that op was not the only one with this issue

Please use the +1 functionality for this in the future. Just click the simely in the top right of the comment to add a +1 to the issue.

**Comment by Xaymar** [comment]
> @meredith2u On The Witcher 3, I have tried limit fps to 60 and turn vsync on, It still happen.
> I'm not sure about your suggest, because I'm using same AMD Driver version and game setting, It only happen on Windows 10 version 1709, version 1607 no issues.

Please check the Task Manager on Version 1709 (Fall Creators Update), it shows CPU and GPU usage. OBS should be slightly using two to four cores, depending on how your scene is built, plus about 15-30% GPU depending on the base and scaled resolution, more during recording/streaming.

**Comment by jonipokka** [comment]
@Xaymar 
I understand the issue but the point I'm trying to make is that **this has never been a problem before**. Something changed when the issue appeared. And again I emphasize on what I said: I'm not pointing a finger at the encoder itself but I thought I would comment on this thread just in case.

I could have used the +1 functionality but as a developer myself I always appreciate more specific information. Even a tiny bit of extra information can be valuable.

Also "now it's time for you to also do your part and balance the load properly". Like I said it is completely impossible to have a modern game run at a constant framerate without **any** kinds of dips.

Anyway I have nothing further to add to this issue and nothing on my end requires fixing etc so I will see myself out now.

**Comment by meredith2u** [comment]
@Xaymar 
Task Manager information:
CPU(E3-1230v2-4C8T) usage 50%~80%, GPU 3D usage is 98%, HEVC recoding encode usage is 52%.
Game: The Witcher 3, Graphics preset high, VSync on, FPS limit 60.

**Comment by Xaymar** [comment]
That looks like heavy GPU congestion, 2% left over is not a lot. Try a clean driver reinstall.

**Comment by meredith2u** [comment]
I will reinstall driver and test again.

**Comment by jonipokka** [comment]
Something is definitely up. Issue started happening again when playing a more hardware demanding game.

I tested recording Destiny 2 and went to a place where fps is known to drop below my refresh rate and did some test recordings.

With AMD Driver version 17.11.2 and OBS version 20.1.3 I lagged quite a bit of frames and got the encoding overloaded error in the log.
[https://drive.google.com/open?id=1JDcqO_-en4GjCPSzdpdofNmq5Uxdeh7T](https://drive.google.com/open?id=1JDcqO_-en4GjCPSzdpdofNmq5Uxdeh7T)

Then I rolled back to AMD Driver 17.7.2 and OBS version 20.1.0 and everything worked fine with no frames dropped and no errors even thou the fps was below the 75 cap in that particular area and GPU was taxed to the limit.
[https://drive.google.com/open?id=11T8RaPgwq9lAYP_doUr3BKwXF_2lfIRD](https://drive.google.com/open?id=11T8RaPgwq9lAYP_doUr3BKwXF_2lfIRD)

**EDIT: The first link was wrong, pointed to the second link. Fixed the links now.**

**Comment by Benman2785** [comment]
https://github.com/Xaymar/obs-studio_amf-encoder-plugin/issues/283

maybe this? but 17.7.2 should also be broken - since it had problems with HWacc

**Comment by meredith2u** [comment]
After my repeated tests, I am convinced that AMD has modified AMF's content, because I switched driver to 17.10.1 no problem, 17.11.1 and 17.11.2 will appear encoding overloaded. I tested on Windows 10 version 1607.
For another reference, A's Video Converter can be used with HEVC at 17.10.1, but not with 17.11.1

**Comment by Benman2785** [comment]
we can ask mikhail about the change of AMF

**Comment by Xaymar** [comment]
To date I have not been able to reproduce this, not with ffmpeg-amf or with the OBS Studio plugin, so I'll need more information from those that do get this problem:

1. What Graphics Card to you have? (I mean the full name, "ASUS Radeon RX 480 ..." and so on).
2. Does it come with a stock overclock? (Are the clocks higher than the reference models?)
3. Which drivers are affected? (Please test all recent with the matching plugin version.)
4. What Windows 10 version are you on? (Win+R, type in `winver`, hit enter, the number after Version and OS Build are the important parts.)
5. Do you have any other Graphics Cards installed? (Internal GPUs are still GPUs!)
6. Finally, the entirety of your Computer Specs: Mainboard, CPU, RAM (+Amount), HDDs, SSDs, Optical Drives, attached USB devices, etc. If you don't know most of these, Speccy by Piriform can help tell you this information. Alternatively you can always upload a `msinfo32` and `dxdiag` file.

**Comment by meredith2u** [comment]
FBirth already ask Mikhail and find the anser, It's should be the driver issues, I will wait when driver fix then close this issue post.

Also, here is information:
1. MSI Radeon RX 480 GAMING X 8G
2. 1303 MHz / 2000 MHz
3. 17.10.1(fine), 17.10.3(overload), 17.11.1(overload), 17.11.2(overload)
4. Windows 10 version 1607, version 1709
5. No.
6. Basic Computer Specs on my first post log.

**Comment by Xaymar** [comment]
The issue FBirth reported is with color conversion, all of the logs here however do not do any color conversion - it is straight up NV12. Edit: Color Conversion is not officially supported anyway and has never been made non-experimental.

**Comment by meredith2u** [comment]
Ok, I thought is the YUV Color Space 709 on OBS setting cause issues(not test 601 yet).

**Comment by jonipokka** [comment]
@Xaymar 
1. Gigabyte Radeon RX 580 AORUS - 4GB (GV-RX580AORUS-4GD)
2. The card comes with a stock overclock
3. AMD Driver version 17.11.2 and plugin version 2.2.4 causes issues. Currently on AMD Driver version 17.7.2 and plugin version 2.2.2 which are working
4. Tested on both 1706 and 1709 versions of Windows 10 Professional (64-bit)
5. I do not have any other graphics cards
6. Specs as follows:
	CPU: Ryzen 5 1600 (Stock, no overclock)
	RAM: 16GB (2 x 8GB) Trident Z, DDR4 3200MHz, 1.35V (F4-3200C14D-16GTZKW)
	MOBO: MSI Mortar B350 motherboard (Latest bios updated)
	PSU: Seasonic 620W M12II-620 Evolution
	GPU: Radeon RX 580 AORUS - 4GB (GV-RX580AORUS-4GD)
	HDD/SSD: 1TB seagate barracuda + 256GB kingston ssdnow
	USB Devices: Creative Livecam HD, Razer Deathadder Elite, Arachnid Recore RGB keyboard, Samson Go Mic
	no optical drives

EDIT: Unfortunately I have a really busy schedule but I will try to take time later today to test out several of the latest drivers. Any specific ones you want tested?

**Comment by meredith2u** [comment]
Today I tested the driver version 17.11.4, the result of the preliminary test seems to have been resolved, I will spend more time to test it, if no problem, I will close the post.

**Comment by meredith2u** [comment]
Unfortunately, I have to reopen this issue, because after several months of repeated testing, I think it should be caused by the driver, I know this is not caused by this plugin, but the issue does exist.

I'm curious why @Xaymar can not reproduce this issue.
I will close this issue when AMD driver fix.

**Comment by Benman2785** [comment]
i think i can relate to this error - with 18.1.1 i also had the bug from #283 again

**Comment by meredith2u** [comment]
@Benman2785 I have seen it and disable pre-pass when I recording The Division using DX12, It's no longer cause crash, but Encoding overloaded still exist.

Your issue already fix on OBS 21.0.1, I have tested it recently.

**Comment by Benman2785** [comment]
@meredith2u nah - still occured on Crimson 18.1.1 and OBS 21.0.1

**Comment by Xaymar** [comment]
Please properly follow the [Troubleshooting Guide](https://github.com/Xaymar/obs-studio_amf-encoder-plugin/wiki/Guide%3A-Troubleshooting) and attach the necessary files.

> Unfortunately, I have to reopen this issue, because after several months of repeated testing, I think it should be caused by the driver, I know this is not caused by this plugin, but the issue does exist.

If the issue is not with the plugin then you are reporting this to the wrong person. I do not work for AMD or am in any way affiliated with AMD, so reporting it to me does not help. A better place would be the AMD Community forum, the AMD Driver error/bug reporter or AMD customer support. "Signal Boosting" won't help get this problem get fixed by AMD if the people at AMD don't know about it.

> I'm curious why @Xaymar can not reproduce this issue.

To put it simply, it's because i leave at least 20% GPU headroom in the most taxing areas of the game, which opens up to ~30-40% GPU headroom in less taxing areas. That left over GPU is then used by a high priority OBS Studio process, which composes, downloads, uploads, encodes and downloads again.

If you do not leave any headroom for OBS Studio to work in you will get lagged frames or encoding lag, since there is not enough time to upload frames to the GPU.

**Comment by meredith2u** [comment]
> To put it simply, it's because i leave at least 20% GPU headroom in the most taxing areas of the game, which opens up to ~30-40% GPU headroom in less taxing areas. That left over GPU is then used by a high priority OBS Studio process, which composes, downloads, uploads, encodes and downloads again.

>If you do not leave any headroom for OBS Studio to work in you will get lagged frames or encoding lag, since there is not enough time to upload frames to the GPU.

I know what you mean, but as far as I know the GPU 3D usage should not affect encoding performance, before driver version 17.10.2 it true, after 17.10.3 it's not.

**Comment by meredith2u** [comment]
>nah - still occured on Crimson 18.1.1 and OBS 21.0.1

@Benman2785 I retested today, It's crashed, sad face.

**Comment by Xaymar** [comment]
Please provide the files asked for in the Troubleshooting Guide. 

> I know what you mean, but as far as I know the GPU 3D usage should not affect encoding performance, before driver version 17.10.2 it true, after 17.10.3 it's not.

Any GPU usage has an effect on GPU encoding in OBS Studio, as there is no pure on-GPU encoding in OBS Studio. This is why comparing it to AMD ReLive or Nvidia ShadowPlay is completely useless, since OBS Studio always downloads the frame to the CPU, while ReLive and ShadowPlay do everything on the GPU.

That means that if you don't leave enough GPU headroom, nothing will work correctly.

**Comment by meredith2u** [comment]
>Any GPU usage has an effect on GPU encoding in OBS Studio, as there is no pure on-GPU encoding in OBS Studio. This is why comparing it to AMD ReLive or Nvidia ShadowPlay is completely useless, since OBS Studio always downloads the frame to the CPU, while ReLive and ShadowPlay do everything on the GPU.

>That means that if you don't leave enough GPU headroom, nothing will work correctly.

I know GPU usage will be effect, but it should be very small, just like I tested QSV ,NVENC, and AMF(17.10.2)  on OBS, it all work fine, not after 17.10.3.

I know this is driver issue, I will wait for the driver to update to 18.xxxx before doing the test.

**Comment by Xaymar** [comment]
The impact on GPU usage depends on how large the frame is. 1920x1080x60 NV12 takes away ~7% GPU for the upload, another ~2-3% if it has to convert to a different color format, some ~1-4% for encode and finally another ~0.2-8% depending on what bitrate is going back.

An empty 1920x1080x60 scene in OBS with NV12 for example takes 9% of the GPU. That is without encoding or anything else, just purely getting that frame back from the GPU into RAM.

**Comment by meredith2u** [comment]
>The impact on GPU usage depends on how large the frame is. 1920x1080x60 NV12 takes away ~7% GPU for the upload, another ~2-3% if it has to convert to a different color format, some ~1-4% for encode and finally another ~0.2-8% depending on what bitrate is going back.

>An empty 1920x1080x60 scene in OBS with NV12 for example takes 9% of the GPU. That is without encoding or anything else, just purely getting that frame back from the GPU into RAM.

Yes, I know GPU encoding will effect fps in game, but recording/streaming video will be smooth, after 17.10.3 it not, I don't think this is normal.

**Comment by Xaymar** [comment]
No, GPU encoding does not affect the in game fps, it is the other way. Unless you use Windows XP or earlier, the focused apps GPU usage is **ALWAYS** prioritized over any other app.

**Comment by meredith2u** [comment]
>No, GPU encoding does not affect the in game fps, it is the other way. Unless you use Windows XP or earlier, the focused apps GPU usage is ALWAYS prioritized over any other app

Ok, but It's still not happen Encoding overloaded on QSV, NVENC and AMF(17.10.2), if it happen, I only need change encode setting to lower encoding load. after 17.10.3 even though I lowered the encoding load, Encoding overloaded always happen.

I'm already report to AMD, hope they will fix this issue.

**Comment by Xaymar** [comment]
@meredith2u Is this still happening for you? If not I will close this report.

**Comment by jonipokka** [comment]
@Xaymar For me it is still happening. AMD Driver version 18.3.2 and OBS 21.1.0 and AMF 2.3.3
Here is a log file for example. On the side note the amount of dropped frames is not as high as it used to be but still a noticeable amount.

[https://pastebin.com/NZnwWDCm](https://pastebin.com/NZnwWDCm)

The weird thing is that this issue does not exist in older version of AMF, OBS and AMD drivers. Like I have said earlier I have absolutely zero problems on AMD driver 17.7.2 and plugin version 2.2.2 with older obs.

**Comment by meredith2u** [comment]
@Xaymar Yes, The new Driver 18.3.3 still happening. Also I have tested Driver 17.10.1 and OBS 21.1.0 and Plugin 2.3.3 has no issues.

**Comment by Benman2785** [comment]
http://benmanshafen.de/obs/xaymar/2.3.3/2018-03-24_11-20-09.txt
http://benmanshafen.de/obs/xaymar/2.3.3/2018-03-24_11-20-09.flv

is that related?

**Comment by Xaymar** [comment]
@Benman2785 Looks like a different issue.

**Comment by Vectrobe** [comment]
This still occurs with 18.3.3 and 4k, any form of capturing other than the preview window is broken and only gets 1-2fps, in many cases it runs full-speed and then drops to 1fps within the first 2 seconds.

Recording at lower than 4k will be perfectly fine however, and the very first recording at 4k after a reboot may come out just fine, with subsequent recordings broken.

This is on a vega64 (4k60 100Mbps) and 1600X (4k30-60, without CPU tasks running).

CL copies used to fix a similar performance issue, but this seems to be unrelated.

**Comment by Xaymar** [comment]
Closing as this is _not_ an issue with the plugin. If you have trouble with drivers or hardware, please contact AMD support as I am powerless to help you no matter what your issue may be.
